remove vague addition store setter filter (#1161)
solidjs/solid-docs: src/routes/concepts/stores.mdx

### Filtering values

To update elements in an array based on specific conditions, you can pass a function as an argument.

> This function will act as a filter, It receives the old value and index, giving you flexibility to apply logic that targets specific cases.

Reviewer: ```suggestion
This function acts as a filter, receiving the old value and index, and gives you the flexibility to apply logic that targets specific cases.
```

setStore("users", (user) => user.username.startsWith("t"), "loggedIn", false)
```

> In addition to [`.startsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith), you can use other array methods like [`.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) to filter for the values that you need.

Reviewer: I can understand the argument for deleting it, however, it was added as a means to address support comments and questions that we got. A solution could be to modify the second sentence to say something like the following:

```text
This function will act as a filter, allowing you to select elements that satisfy the condition using logic that may involve methods like `.startsWith()`, `includes()`.
```

What are your thoughts on that?

Author: removing the `Array.find` was my main objective 
I think your suggestion is good, but if you look at the full context it has some repetition.  

below is the full context for better reference 
and I have added some examples that should cover common use cases. 

so perhaps it can be a good substitution for the "additional" part we removed. 

---
https://docs.solidjs.com/concepts/stores#filtering-values

## [Filtering values](https://docs.solidjs.com/concepts/stores#filtering-values)
In scenarios where you want to update elements in an array based on a specific condition, you can pass a function as an argument. This function will act as a filter, allowing you to select elements that satisfy the condition. It receives the old value and index as arguments, providing the flexibility to make conditional updates.

```js
// update users with username that starts with "t"
setStore("users", (user) => user.username.startsWith("t"), "loggedIn", false)

// update users with username "abc"
setStore("users", (user) => user.username == "abc" , "loggedIn", false)

// update users with id 1, 2 or 3 
let ids = [1,2,3] 
setStore("users", (user) => ids.includes(user.id) , "loggedIn", false)
```

Reviewer: I can see where you're going when you want to introduce some more examples, however, I prefer an explicit callout because code blocks shouldn't be used in place of "learning", if that makes sense. Just as images should supplement rather than replace the explanation, code blocks should reinforce what's already been explained, not serve as the primary means of conveying new concepts, hence why I had suggested introducing changes to the second sentence.

Perhaps this edit below, coupled with your updated example, might work better:

To update elements in an array based on specific conditions, you can pass a function as an argument. It receives the old value and index, giving you flexibility to apply logic that targets specific cases. This might include using methods like `.startsWith()`, `includes()`, or other comparison techniques to determine which elements should be updated.

Author: >  I prefer an explicit callout because code blocks shouldn't be used in place of "learning"

I strongly agree with you on this one 
I recently noticed this in another PR and made a suggestion to unpack 
the explanation from the code.

https://github.com/solidjs/solid-docs/pull/1163/files#diff-2c9de02907fec849b5feed69bdc4b87f7b420ee9585a26fc6cf8b9bdf7ef9c9bR79

there is another issue, I realized with comments in code, is that they are not 
fully visible due to scrolling if they are too long. this perhaps can be addressed in another issue  / guidelines 
for contributors to pay attention to. 

--- 

In regard to this PR , do you suggest replacing the **current** with **suggested**

|suggested @LadyBluenotes  | current |
|---|---|
| To update elements in an array based on specific conditions, you can pass a function as an argument. It receives the old value and index, giving you flexibility to apply logic that targets specific cases. This might include using methods like .startsWith(), includes(), or other comparison techniques to determine which elements should be updated.  |  In scenarios where you want to update elements in an array based on a specific condition, you can pass a function as an argument. This function will act as a filter, allowing you to select elements that satisfy the condition. It receives the old value and index as arguments, providing the flexibility to make conditional updates.  |  

I am happy with your change of copy and the supplementary code examples. 

I will suggest a minor modification to your recent suggestion ( see in bold ) : 

||
|-|
|To update elements in an array based on specific conditions, you can pass a function as an argument. **This function will act as a filter**, It receives the old value and index, giving you flexibility to apply logic that targets specific cases. This might include using methods like .startsWith(), includes(), or other comparison techniques to determine which elements should be updated.|

If all is good, I can commit the change
